### ISSUE TITLE
Linter issues "This else clause can be omitted." when not applicable

### ISSUE THREAD

**Issue by ritsaert** [biomejs/biome#1060]
### Environment information

```bash
The linter rule noUselessElse triggers to aggressively, even when the else clause is indeed not useless.
If seen for face value you might delete useful code.
```

### Rule name

lint/style/noUselessElse

### Playground link

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAaABlAGwAbABvAFcAbwByAGwAZAAoAGIAOgAgAGIAbwBvAGwAZQBhAG4AKQAgAHsACgAgACAAaQBmACAAKABiACkAIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgADEAOwAKACAAIAB9ACAAZQBsAHMAZQAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAMgA7AAoAIAAgAH0ACgB9AA%3D%3D

### Expected result

Linter rule should not be triggered

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Hi!

The `else` clause of your playground example is actually useless.
Indeed, your example:

```js
function helloWorld(b: boolean) {
  if (b) {
    return 1;
  } else {
    return 2;
  }
}
```

can be simplified to:

```js
function helloWorld(b: boolean) {
  if (b) {
    return 1;
  }
  return 2;
}
```

**Comment by veloc1** [comment]
Got a bit different case with `noUselessElse` and biome version 1.4.0:

```typescript
 const data: { f1: string } | undefined = { f1: "foobar" };
  if (!data) {
    //
  } else if (data && data.f1 === "foo") {
    return;
  } else if (data && data.f1 === "baz") {
  }
  ```
  
  Second `else if` triggers `lint/style/noUselessElse`
  
  Fixed by updating to 1.4.1 (#943)

**Comment by Conaclos** [comment]
Yes, we changed a bit the behavior of `noUselessElse` in the 1.4.1 version to accept code like yours.

**Comment by ritsaert** [comment]
Nice. I think the main reason I was triggered by the behaviour is the linter message "This else clause can be omitted." instructed me to remove the whole else clause altogether.  Whereas the statements within the else clause are clearly relevant.

**Comment by Conaclos** [comment]
Good point. Have you some suggestions to improve the message?

**Comment by Conaclos** [comment]
The message was improved to `This else clause can be omitted because previous branches break early.`. Thanks for your help :)
